src/client/kv/dc_kv.c

```diff
@@ -138,31 +140,32 @@ dc_kv_get(tse_task_t *task)
 {
 	daos_kv_get_t		*args = daos_task_get_args(task);
 	daos_obj_fetch_t	*fetch_args;
-	tse_task_t		*fetch_task;
-	struct io_params	*params;
+	tse_task_t		*fetch_task = NULL;
+	struct io_params	*params = NULL;
 	void			*buf;
 	daos_size_t		*buf_size;
 	int			rc;
 
+	if (args->key == NULL)
+		D_GOTO(err_task, rc = -DER_INVAL);
+
 	buf = args->buf;
 	buf_size = args->buf_size;
 	if (buf_size == NULL) {
 		D_ERROR("Buffer size pointer is NULL\n");
-		return -DER_INVAL;
+		D_GOTO(err_task, rc = -DER_INVAL);
 	}
 
 	D_ALLOC_PTR(params);
-	if (params == NULL) {
-		D_ERROR("Failed memory allocation\n");
-		return -DER_NOMEM;
-	}
+	if (params == NULL)
+		D_GOTO(err_task, rc = -DER_NOMEM);
 
 	/** init dkey */
 	d_iov_set(&params->dkey, (void *)args->key, strlen(args->key));
 
-	/** init iod. For now set akey = dkey */
-	d_iov_set(&params->iod.iod_name, (void *)args->key,
-		     strlen(args->key));
+	/** init iod. */
+	params->akey_str = '0';
```

Comment:
Nit-picking: akey_str is a char, not a string. &akey_str also not a string for the lack of '\0'. Reminds me of a C language test. :)